src/Synthesiser.cpp

```diff
@@ -130,20 +130,21 @@ void Synthesiser::generateRelationTypeStruct(
     relationType->generateTypeStruct(out);
 }
 
-/* Convert SearchColums to a template index */
+/* Convert SearchColumns to a template index */
 std::string Synthesiser::toIndex(SearchSignature key) {
     std::stringstream tmp;
     tmp << "<";
-    int i = 0;
-    while (key != 0) {
-        if ((key % 2) != 0u) {
-            tmp << i;
-            if (key > 1) {
+    bool first = true;
+
+    for (size_t i = 0; i < key.arity(); ++i) {
+        if (key[i]) {
+            if (!first) {
                 tmp << ",";
+            } else {
+                first = false;
             }
+            tmp << i;
         }
-        key >>= 1;
-        i++;
     }
```

Comment:
Simplify: `tmp << join(key); if (!key.empty()) tmp << ",";`
(Matching proposed change. IDK if you really need/want that tailing `,` if non-empty.)

Reply:
This wasn't called anywhere and I just use the overloaded ostream operator to print the SearchSignature in these contexts (such as lowerUpperRange_x etc.) therefore I just deleted it. Good point for future reference that format() is the more efficient idiom.